### PR TITLE
Update python detection in codegen

### PR DESCRIPTION
**Pull request by sagudev**
Should fix https://github.com/servo/servo/pull/31844#issuecomment-2019685686

---

- [ ] `./mach build -d` does not report any errors
- [ ] `./mach test-tidy` does not report any errors
- [ ] These changes fix #___ (GitHub issue number if applicable)

- [ ] There are tests for these changes OR
- [ ] These changes do not require tests because ___

### CODE REVIEW

**Review by mukilan — APPROVED** [review]
Thanks for this fix!

**Review by mrobinson — APPROVED** [review]
Nice catch!

File: components/script/build.rs [hunk 1]
```diff
@@ -67,9 +67,9 @@ impl<'a> phf_shared::PhfHash for Bytes<'a> {
 fn find_python() -> String {
     env::var("PYTHON3").ok().unwrap_or_else(|| {
         let candidates = if cfg!(windows) {
-            ["python3.8.exe", "python38.exe", "python.exe"]
+            ["python.exe"]
```
[the remark this answers is not in the document]
Comment (sagudev) [reply components/script/build.rs:70]: I have installed python using python.org installer but it only provides python.exe. `python3` opens MS store with another python to install (this python version is somehow different then the one provided by python.org as some pip packages fail to install).

**Comment by eerii** [comment]
Thank you for fixing this!

**Pull request merged**